src/components/NcNoteCard/NcNoteCard.vue

```diff
@@ -0,0 +1,156 @@
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<docs>
+This component is made to display additional information to the user. It is
+available in three versions:
+
+- success: Display a successful message
+- warning: Display a warning to the user. This indicate that the action they want
+- error: Display an error message. For example
+
+When using an error type,
+
+```vue
+<div>
+	<NcNoteCard type="warning">
+		<p>This is dangerous</p>
+	</NcNoteCard>
+
+	<NcNoteCard type="error" heading="Error">
+		<p>The server is not happy and reported the following error</p>
+	</NcNoteCard>
+
+	<NcNoteCard type="success">
+		<p>You won</p>
+	</NcNoteCard>
+</div>
+```
+</docs>
+
+<template>
+	<div class="notecard"
+		:class="`notecard--${type}`"
+		:role="shouldShowAlert ? 'alert' : ''">
+		<component :is="icon"
+			class="notecard__icon"
+			:class="{'notecard__icon--heading': heading}"
+			:fill-color="color" />
+		<div>
+			<h2 v-if="heading">
+				{{ heading }}
+			</h2>
+			<slot />
+		</div>
+	</div>
+</template>
+
+<script>
+import CheckboxMarkedCircle from 'vue-material-design-icons/CheckboxMarkedCircle.vue'
+import AlertDecagram from 'vue-material-design-icons/AlertDecagram.vue'
+import Alert from 'vue-material-design-icons/Alert.vue'
+
+export default {
+	name: 'NcNoteCard',
+
+	props: {
+		/**
+		 * Type of the attribute
+		 */
+		type: {
+			type: String,
+			default: () => 'warning',
+			validator: type => type === 'success' || type === 'warning' || type === 'error',
+		},
+		showAlert: {
+			type: Boolean,
+			default: () => false,
+		},
+		heading: {
+			type: String,
+			required: false,
+		},
+	},
+	computed: {
+		shouldShowAlert() {
+			return this.showAlert || this.type === 'error'
+		},
+		icon() {
+			switch (this.type) {
+			case 'error':
+				return AlertDecagram
+			case 'success':
+				return CheckboxMarkedCircle
+			case 'warning':
+				return Alert
+			default:
+				return Alert
+			}
+		},
+		color() {
+			switch (this.type) {
+			case 'error':
+				return 'var(--color-error)'
+			case 'success':
+				return 'var(--color-success)'
+			case 'warning':
+				return 'var(--color-warning)'
+			default:
+				return 'var(--color-warning)'
+			}
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+.notecard {
+	color: var(--color-text-light) !important;
+	background-color: var(--note-background) !important;
+	border-left: 4px solid var(--note-theme);
```

Comment:
```suggestion
	border-inline-start: 4px solid var(--note-theme);
```
If you want to support rtl languages.